File: BinaryTree.java
import java.util.Scanner;
class Node{
    int data;Node left,right;
    public Node(int data){
        this.data=data;
        left=right=null;
    }
}
public class BinaryTree {
    Node root;
    int level;
    public BinaryTree() {
        root = null;
    }
    public void insert(int data){
        root=insertRec(root,data);
    }
    private Node insertRec(Node root,int data){
        if (root == null) {
            root = new Node(data);
            return root;
        }
        if (data < root.data){
            root.left = insertRec(root.left, data);
        }
        else if (data > root.data){
            root.right = insertRec(root.right, data);
        }
        return root;
    }
    public void show(){
        System.out.println("The Tree with sorted values: ");
        showRec(root);
    }
    public void showRec(Node root){
        if(root!=null){
            showRec(root.left);
            System.out.print(root.data + " ");
            showRec(root.right);
        }
    }
    private Node deleterec(Node root,int data){
        if(root==null){
            System.out.println("Tree is Empty");
            return root;
        }
        if(data<root.data){
            root.left=deleterec(root.left, data);
        }
        else if(data<root.data){
            root.right=deleterec(root.right, data);
        }
        else{
            if(root.left==null && root.right==null){
                root=null;
                return root;
            }
            else if(root.left==null){
                Node temp = root.right;
                root=null;
                return temp;
            }
            else if(root.right==null){
                Node temp = root.left;
                root=null;
                return temp;
            }
            else {
                while(root.left!=null){
                    root=root.left;
                }
                Node temp=root;
                root.data=temp.data;
                root.right=deleterec(root.right, data);
            }
        }
        return root;
    }
    public void del(int data){
        if(root==null){
            System.out.println("Tree is Empty!");
        }
        root=deleterec(root,data);
        System.out.println("Element Deleted!");
    }
    public static void main(String[] args) {
        BinaryTree tree=new BinaryTree();
        int ch,data;
        do{
            System.out.println("\nMenu");
            System.out.println("1. Insert ");
            System.out.println("2. Show ");
            System.out.println("3. Delete");
            System.out.println("4. Exit ");
            System.out.println("Enter your choice: ");
            Scanner sc = new Scanner(System.in);
            ch= sc.nextInt();
            switch (ch) {
                case 1:
                    System.out.println("Enter the Value to be Inserted: ");
                    data=sc.nextInt();
                    tree.insert(data);
                    break;
                case 2: tree.show();break;
                case 3: 
                    System.out.println("Enter the Value to be Deleted: ");
                    data=sc.nextInt();
                    tree.insert(data);
                    tree.del(data);
                case 4: System.out.println("Menu Exited");break;
                default: System.out.println("Invalid Choice ");
                sc.close();
            }
        }
        while(ch!=4);
    }
}

File: binarytree.cpp

#include <iostream>
using namespace std;
class Node{  
    public:
    int data;
    Node*left;
    Node*right;
    Node(){
        data=0;
        left=NULL;
        right=NULL;
    }
    Node(int data){
        this->data=data;
        this->left=NULL;
        this->right=NULL;
    }
};
class binarytree{
    Node*root;
    public:
    binarytree(){
        root=NULL;
    }    
    void insert(int data){
        root= insertrec(root,data);
    }
    Node*insertrec(Node*root,int data){
        if(root==nullptr){
            root=new Node(data);
            return root;
        } 
        if(data<root->data){
            root->left=insertrec(root->left,data);
        }
        else if(data>root->data){
            root->right=insertrec(root->right,data);
        }
        return root;
    }
    void show(){
        cout<<"The Tree with sorted values: \n";
        showrec(root);
    }
    void showrec(Node*root){
        if(root!=nullptr){
            showrec(root->left);
            cout<<root->data<<" \n";
            showrec(root->right);
        }
    }
    void del(int data){
        if(root==nullptr){
            cout<<"Tree is Empty!\n";
            return;
        }
        root=deleterec(root,data);
        cout<<"Node Deleted!";
    }
    Node*findMin(Node*root){
        while(root->left!=nullptr){
            root=root->left;
        }
        return root;
    }
    Node*deleterec(Node*root,int data){
        if(root==nullptr){
            cout<<"Tree is Empty!";
            return root;
        }
        if(data<root->data){
            root->left=deleterec(root->left,data);
        }
        else if(data>root->data){
            root->right=deleterec(root->right,data);
        }
        else{
            if(root->left==nullptr && root->right==nullptr){
                delete root;
                return nullptr;
            }
            else if(root->left==nullptr){
                Node*temp=root->right;
                delete root;
                return temp;
            }
            else if(root->right==nullptr){
                Node*temp=root->left;
                delete root;
                return temp;
            }
            else{
                Node*temp=findMin(root->right);
                root->data=temp->data;
                root->right=deleterec(root->right,temp->data);
            }
        }
        return root;
    }
    void search(int data){
        Node*curr=root;
        while(curr!=nullptr){
            if(data==curr->data){
                cout<<"Element found!\n";
                return;
            }
            else if(data<curr->data){
                curr=curr->left;
            }
            else{
                curr=curr->right;
            }
        }
        cout<<"Element not found!\n";
    }
};
int main(){
    binarytree bin;
    int ch,dt;
    do{
        cout<<"\nMenu\n";
        cout<<"1. Insertion\n"<<"2. Traversal\n"<<"3. Delete\n"<<"4. Search\n"<<"5. Exit\n"<<"Enter your choice: ";
        cin>>ch;
        switch(ch){
            case 1: cout<<"Enter the value to be inserted: ";cin>>dt; bin.insert(dt);break;
            case 2: bin.show();break;
            case 3: cout<<"Enter the value to be deleted: ";cin>>dt; bin.del(dt);break;
            case 4: cout<<"Enter the value to be searched: ";cin>>dt; bin.search(dt);break;
            case 5: cout<<"Menu exited!";break;
            default: cout<<"Invalid Case!\n";
        }
    }
    while(ch!=5);
    return 0;
}
